### PR TITLE
Showcase: add custom 500 error page

### PR DESCRIPTION
**Pull request by maxime-poulain**
1e0fe692dddf655c65bc7271c117829f0238eafd introduced error page for the UserAdmin project but not for the showcase one.
This commit brings the same page for the showcase.

https://smartjira.atlassian.net/browse/CFA-506

### CODE REVIEW

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/500.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        else
+        {
+            ErrorTitle = ShowcaseResources.GenericNotFoundPageTitle;
+        }
+    }
+
+    private void SetMessage()
+    {
+        if (TempData[ErrorMessageKey]?.ToString() == string.Empty)
```
Comment (LievenPraet) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/500.cshtml.cs:41]: We can use `String.IsNullOrWhiteSpace` to check the value

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Program.cs [hunk 2]
```diff
@@ -21,16 +21,15 @@
 // A fix which allows SQL domain authentication with Kerberos on linux host
 NetSecurityNativeFix.Initialize(app.Services.GetRequiredService<ILogger<Program>>());
 
-app.UseStatusCodePagesWithReExecute("/{0}");
-
 // Configure the HTTP request pipeline.
-if (!app.Environment.IsDevelopment())
+if (app.Environment.IsProduction())
 {
-    app.UseExceptionHandler("/Error");
```
Comment (LievenPraet) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Program.cs:29]: Do we still need the `/Error` endpoint if it is no longer referenced here?

Comment (maxime-poulain) [reply src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Program.cs:29]: Error end point is deleted (Error.cshtml).

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/500.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace Smart.FA.Catalog.Showcase.Web.Pages;
+
+public class ServerErrorPage : PageModel
+{
+    public const string ErrorTitleKey = nameof(ErrorTitle);
+
+    public const string ErrorMessageKey = nameof(ErrorMessageKey);
```
[the remark this answers is not in the document]
Comment (maxime-poulain) [reply src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/500.cshtml.cs:9]: I assume it should be nameof(ErrorTitleKey).
I will look globally if the key is hardcoded.

I will need to adjust accordingly the other 500 page I copy the content from.

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/500.cshtml.cs (hunk 1 above)
Comment (maxime-poulain) [reply src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/500.cshtml.cs:41]: I ll proceed to change the source page also.

**Review by LievenPraet — APPROVED** [review]

**Pull request merged**